ssz_serialization/merkleization.nim

```diff
@@ -631,7 +652,7 @@ template chunkedHashTreeRoot[T](
 
 func bitListHashTreeRoot(
     merkleizer: var SszMerkleizerImpl, x: BitSeq,
-    chunks: Slice[Limit]): Digest =
+    chunks: Slice[Limit]): Digest {.noinit.} =
```

Comment:
There `{.noinit.}` on return values are strange too, similar to the `let` ones, for analogous reasons. Nim knows if it's doing a `result` return (semantically) and if the return path always has some value, then same comment as the `let`.

Reply:
in theory, theory and practice are the same...